tests/rustdoc/intra-doc/deps.rs

```diff
@@ -0,0 +1,22 @@
+// Checks that the link generated
+// Regression test for <https://github.com/rust-lang/rust/issues/137857>.
+
+//@ compile-flags: --document-private-items -Z unstable-options
+//@ compile-flags: --extern-html-root-url=empty=https://docs.rs/
+// This one is to ensure that we don't link "by default" any item we see which has
+// an external html root URL.
+//@ compile-flags: --extern-html-root-url=non_existent=https://docs.rs/
+//@ aux-build: empty.rs
+
+#![crate_name = "foo"]
+#![allow(rustdoc::broken_intra_doc_links)]
```

Comment:
```suggestion
#![expect(rustdoc::broken_intra_doc_links)]
```

Another layer to ensure we're not linking to `non_existant`